cmd/gateway/server.go

```diff
@@ -106,3 +111,53 @@ func azFn(c *gin.Context) {
 	c.JSON(http.StatusOK, gin.H{"message": "OK"})
 	return
 }
+
+// ceFn is a cloud events handler.
+func ceFn(c *gin.Context) {
```

Comment:
The first thing we're going to need here is endpoint validation - https://docs.microsoft.com/en-us/azure/event-grid/security-authentication


When creating the event subscription, the first event will be a `Microsoft.EventGrid.SubscriptionValidationEvent` (which does not appear to be CloudSpec compliant?)

```
[
    {
        "id": "be0d5d8f-585b-4ace-861f-fc4beaaf2477",
        "topic": "/subscriptions/<subscription-id>/resourceGroups/eventgrid-cloudevents/providers/microsoft.storage/storageaccounts/evcloudeventsstorage",
        "subject": "",
        "data": {
            "validationCode": "validation-code",
            "validationUrl": "https://rp-northeurope.eventgrid.azure.net/eventsubscriptions/local-ngrok/validate?id=validation-code&t=2018-05-25T11:41:08.7774182Z&apiVersion=2018-05-01-preview&token=token"
        },
        "eventType": "Microsoft.EventGrid.SubscriptionValidationEvent",
        "eventTime": "2018-05-25T11:41:08.8086405Z",
        "metadataVersion": "1",
        "dataVersion": "2"
    }
]
```

It expects a response of type:

```
{
    "validationResponse": "validation-code"
}
```

Reply:
Yes, it appears that the validation event does not follow the CloudEvents spec, but the EventGrid default schema. So when we receive a request, we need to decode it into an EventGrid type - if that fails, try to decode it as CloudSpec event.